posthog/models/dashboard_tile.py

```diff
@@ -3,20 +3,39 @@
 from django.db import models
 from django.db.models.signals import post_save
 from django.dispatch import receiver
+from django.utils import timezone
 
 from posthog.models.dashboard import Dashboard
 from posthog.models.insight import Insight, generate_insight_cache_key
 
 
-class DashboardTile(models.Model):
-    # Relations
-    dashboard = models.ForeignKey("posthog.Dashboard", on_delete=models.CASCADE)
-    insight = models.ForeignKey("posthog.Insight", on_delete=models.CASCADE)
-
-    # Dashboard layout and style
+class BaseDashboardTile(models.Model):
     layouts: models.JSONField = models.JSONField(default=dict)
     color: models.CharField = models.CharField(max_length=400, null=True, blank=True)
 
+    class Meta:
+        abstract = True
+
+
+class DashboardTextTile(BaseDashboardTile):
+    dashboard = models.ForeignKey("posthog.Dashboard", on_delete=models.CASCADE, related_name="text_tiles")
+    body: models.CharField = models.CharField(max_length=4000, null=True, blank=True)
+    created_by: models.ForeignKey = models.ForeignKey("User", on_delete=models.SET_NULL, null=True, blank=True)
+    last_modified_at: models.DateTimeField = models.DateTimeField(default=timezone.now)
+    last_modified_by: models.ForeignKey = models.ForeignKey(
+        "User", on_delete=models.SET_NULL, null=True, blank=True, related_name="modified_text_tiles"
+    )
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["dashboard"], name="query_by_dashboard_idx"),
+        ]
```

Comment:
Django creates indexes for each `ForeignKey` field automatically, so this seems redundant.